gen/moon.hoon

```diff
@@ -10,8 +10,11 @@
         $~
         $~
     ==
-:-  %noun
-?>  =(1 (met 5 p.bec))
+:-  %tang  :_  ~  :-  %leaf
+=+  len=(met 5 p.bec)
+?.  (lte len 1)
+  =+  typ=?:(=(2 len) "moon" "comet")
+  "can't create a moon from a {typ}"
 =+  mon=(mix (lsh 5 1 (end 5 1 eny)) p.bec)
```

Comment:
This might be golfing a little, but I've been reading src and came across `++clan` and `++rank`...

Would this be more idiomatic ?

```
=+  ran=(clan p.bec)
?:  ?=({?($earl $pawn)} ran)
  "can't create a moon from a {?:(?=($earl ran) "moon" "comet")}"
```